synapse/events/spamcheck.py

```diff
@@ -17,11 +17,12 @@
 import inspect
 from typing import Dict
 
+import synapse.server
```

Comment:
beware that this can often lead to an import cycle, since `synapse.server` imports *everything*. I guess you've gotten away with it here somehow...

Reply:
Yes, this was giving me trouble. @anoadragon453 suggested putting the type in quotes below and that helped to fix this, but it still seemed odd to need to import this here. Is there any fix for this?

Reply:
something like this, possibly:

```
MYPY = False
if MYPY:
    import synapse.server
```

see https://mypy.readthedocs.io/en/stable/common_issues.html#import-cycles.

Reply:
👍 Nice. I knew I had read that somewhere and couldn't find it again on the [imports page](https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports)!

I made this change so that it won't randomly break in the future.